File: FrontEnd/src/app/shared/user.model.ts
import { from } from "rxjs";
import {tick} from '@angular/core/testing';

export class User {

    Id :number; 
    FirstName :string;
    LastName :string;
    Email :string;
    UserName :string;
    Password :string;
    SuperUser :boolean;

    constructor(id:number,firstname:string,lastname:string,email:string,username:string,password:string,superuser:boolean){
        this.Id=id;
        this.FirstName=firstname;
        this.LastName=lastname;
        this.Email=email;
        this.UserName=username;
        this.Password=password;
    }
}

File: BackEndUserRegistration/BackEndUserRegistration/Models/UserModel.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace BackEndUserRegistration.Models
{
    public class UserModel
    {
        //[Key]
        //public int Id { get; set; }
        [Required]

        public string FirstName { get; set; }

        public string LastName { get; set; }
        [Required]

        public string Email { get; set; }
        [Required]

        public string Username { get; set; }
        [Required]

        public string Password { get; set; }

        public bool SuperUser { get; set; }

        public string Role { get; set; }
    }
}
